Gems/ROS2/Code/Source/Manipulator/MotorizedJointComponent.cpp

```diff
@@ -0,0 +1,275 @@
+                        "SinusoidalTest",
+                        "Allows to apply sinusoidal test signal")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJointComponent::m_sinAmplitude,
+                        "Amplitude",
+                        "Amplitude of sinusoidal test signal.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJointComponent::m_sinFreq,
+                        "Frequency",
+                        "Frequency of sinusoidal test signal.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJointComponent::m_sinDC, "DC", "DC of sinusoidal test signal.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJointComponent::m_debugPrint, "Debug", "Print debug to console")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJointComponent::m_measurementReferenceEntity,
+                        "Step Parent",
+                        "Allows to override a parent to get correct measurement");
+            }
+        }
+    }
+    void MotorizedJointComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        const float measurement = ComputeMeasurement(time);
+        if (m_testSinusoidal)
+        {
+            m_setpoint = m_sinDC + m_sinAmplitude * AZ::Sin(m_sinFreq * time.GetSeconds());
+        }
+        const float control_position_error = (m_setpoint + m_zeroOffset) - measurement;
+        m_error = control_position_error; // TODO decide if we want to expose this control error.
+
+        if (m_debugDrawEntity.IsValid())
+        {
+            if (m_linear)
+            {
+                AZ::Transform transform = AZ::Transform::Identity();
+                transform.SetTranslation(m_jointDir * (m_setpoint + m_zeroOffset));
+                AZ::TransformBus::Event(
+                    m_debugDrawEntity, &AZ::TransformBus::Events::SetLocalTM, transform * m_debugDrawEntityInitialTransform);
+            }
+            else
+            {
+                AZ_Assert(false, "Not implemented");
+            }
+        }
+
+        const uint64_t deltaTimeNs = deltaTime * 1'000'000'000;
+        float speed_control = m_pidPos.ComputeCommand(control_position_error, deltaTimeNs);
+
+        if (measurement <= m_limits.first)
+        {
+            // allow only positive control
+            speed_control = AZStd::max(0.f, speed_control);
+        }
+        else if (measurement >= m_limits.second)
+        {
+            // allow only negative control
+            speed_control = AZStd::min(0.f, speed_control);
+        }
+
+        if (m_debugPrint)
+        {
+            AZ_Printf(
+                "MotorizedJointComponent",
+                " %s | pos: %f | err: %f | cntrl : %f | set : %f |",
```

Comment:
`AZ_Printf` doesn't come with a newline automatically.
One needs to be added to have the next output using the stdout stream to be on a newline.
```suggestion
            AZ_Printf(
                "MotorizedJointComponent",
                " %s | pos: %f | err: %f | cntrl : %f | set : %f |\n",
```